ractor.c

```diff
@@ -927,7 +927,16 @@ ractor_select(rb_execution_context_t *ec, const VALUE *rs, int alen, VALUE yield
         ractor_basket_setup(ec, &cr->wait.yielded_basket, yielded_value, move, false);
     }
 
-    // TODO: shuffle actions
+    // Shuffle actions
```

Comment:
I think the round Robbin is enough.
what do you think about it?

Reply:
I'm not quite sure I understand round robin you are mentioning. Could you explain a little bit more? In current implementation, select always pick the first ready ractor, and it easily leads to the situation that the next ractors in the select statement never get a chance to be selected.

Btw, I see that you mentioned Golang is using round-robin technique. But from Golang's specs (https://golang.org/ref/spec#Select_statements), and the implementation (https://golang.org/src/runtime/select.go), the polling order is randomly shuffle.

Reply:
introduce `static int start` and increment this variable at every loop in `ractor_select()`.
use `rs[(n+start)%rs_size]` instead of `rs[(n+start)]`.